pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/PanDomainAuthSettingsRefresher.scala

```diff
@@ -1,83 +1,61 @@
+import org.slf4j.LoggerFactory
 
-import scala.concurrent.duration.FiniteDuration
+import scala.language.postfixOps
 
 /**
   * PanDomainAuthSettingsRefresher will periodically refresh the pan domain settings and expose them via the "settings" method
   *
-  * @param domain the domain you are authin agains
+  * @param domain the domain you are authenticating against
   * @param system the identifier for your app, typically the same as the subdomain your app runs on
   * @param bucketName the bucket where the settings are stored
-  * @param actorSystem the actor system to create the refresh actor
-  * @param awsCredentialsProvider AWS credential provider
-  * @param awsRegion AWS region
-  * @param proxyConfiguration optional proxy configuration
+  * @param settingsFileKey the name of the file that contains the private settings for the given domain
+  * @param s3Client the AWS S3 client that will be used to download the settings from the bucket
+  * @param scheduler optional scheduler that will be used to run the code that updates the bucket
   */
 class PanDomainAuthSettingsRefresher(
   val domain: String,
   val system: String,
-  bucketName: String,
-  actorSystem: ActorSystem,
-  awsCredentialsProvider: AWSCredentialsProvider,
-  awsRegion: Regions,
-  proxyConfiguration: Option[ProxyConfiguration] = None
+  val bucketName: String,
+  settingsFileKey: String,
+  val s3Client: AmazonS3,
+  scheduler: ScheduledExecutorService = Executors.newScheduledThreadPool(1)
 ) {
-  lazy val bucket = new S3Bucket(bucketName, awsCredentialsProvider, awsRegion, proxyConfiguration)
-
-  private lazy val settingsMap = bucket.readDomainSettings(domain)
-  private lazy val authSettings: Agent[PanDomainAuthSettings] = Agent(PanDomainAuthSettings(settingsMap))
-
-  private lazy val domainSettingsRefreshActor = actorSystem.actorOf(Props(classOf[DomainSettingsRefreshActor], domain, bucket, authSettings), "PanDomainAuthSettingsRefresher")
-
-  actorSystem.scheduler.scheduleOnce(1 minute, domainSettingsRefreshActor, Refresh)
-
-  def settings = authSettings.get()
-}
-
-class DomainSettingsRefreshActor(domain: String, bucket: S3Bucket, authSettings: Agent[PanDomainAuthSettings]) extends Actor {
+  private val logger = LoggerFactory.getLogger(this.getClass)
 
-  val frequency: FiniteDuration = 1 minute
-  val log = Logging(context.system, this)
+  // This is deliberately designed to throw an exception during construction if we cannot immediately read the settings
+  private val authSettings: AtomicReference[PanDomainAuthSettings] = new AtomicReference[PanDomainAuthSettings](loadSettings() match {
+    case Right(settings) => PanDomainAuthSettings(settings)
+    case Left(err) => throw Settings.errorToThrowable(err)
+  })
 
-  override def receive: Receive = {
-    case Refresh => {
-      try {
-        val settingsMap = bucket.readDomainSettings(domain)
+  scheduler.scheduleAtFixedRate(() => refresh(), 1, 1, TimeUnit.MINUTES)
```

Comment:
This refresh time was not customisable by calling code before, but it is for the public settings refresher! Perhaps we should also fix in this release?